src/wp-includes/class-wp-comment-query.php

```diff
@@ -451,9 +451,9 @@ public function get_comments() {
 		$key          = md5( serialize( $_args ) );
 		$last_changed = wp_cache_get_last_changed( 'comment' );
 
-		$cache_key   = "get_comments:$key:$last_changed";
+		$cache_key   = "get_comments:$key";
 		$cache_value = wp_cache_get( $cache_key, 'comment-queries' );
-		if ( false === $cache_value ) {
+		if ( ! wp_is_valid_last_changed( $cache_value, $last_changed ) ) {
```

Comment:
I'm not sure about using the word "valid". Any last changed time is a "valid" time, but it might mean the cache should be considered expired.

Maybe we should rename the function to something like `wp_is_cache_value_last_changed( $cache_value, $last_changed )`? That IMO clarifies the purpose better, especially in combination with its parameters.